src/fetch/FetchExample.re

```diff
@@ -0,0 +1,93 @@
+type tDog = string;
+
+type tArrayDogs = array(tDog);
+
+type state = { dogs: tArrayDogs, isLoading: bool, error: bool };
```

Comment:
you might want to consider using a variant type instead using multiple bool fields. For example:

```ml
type state =
| Initial
| Loading
| Error
| Loaded(array(dogs));
```

That'll make it easier to ensure you're actually dealing with all these states.

Reply:
Sounds great, how can we set the state with variants? I tried to find it with in documentation but I couldn't.

Reply:
Just return it in `ReasonReact.Update` as usual:

```ml
ReasonReact.Update(Error);
```

or

```ml
ReasonReact.Update(Loaded(dogs));
``